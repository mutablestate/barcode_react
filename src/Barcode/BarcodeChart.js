import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import { barStyle } from '../utils/barStyle';

const Bar = ({ bar, selectedColor }) => {
  const style = { ...barStyle({ bar }), backgroundColor: selectedColor };
  return (
    <div style={style} data-test={`BarcodeChart:Bar:div:${selectedColor}`}>
      {bar}
    </div>
  );
};

// BarcodeChart
const BarcodeChart = ({
  bars,
  checksumDigit,
  primaryColor,
  secondaryColor
}) => {
  const hasPrimaryBars = bars.length > 0;
  const primary = hasPrimaryBars
    ? bars.map((bar, index) => (
        <Bar key={index} bar={bar} selectedColor={primaryColor} />
      ))
    : null;
  const secondary = hasPrimaryBars ? (
    <Bar bar={checksumDigit} selectedColor={secondaryColor} />
  ) : null;

  return (
    <Fragment>
      <h4 className="align-center">Generated Barcode</h4>
      <div className="flex-container">
        {primary}
        {secondary}
      </div>
    </Fragment>
  );
};

BarcodeChart.propTypes = {
  /** The list of bars */
  bars: PropTypes.array.isRequired,
  /** The generated checksum digit */
  checksumDigit: PropTypes.number.isRequired,
  /** The background color for the primary bars (input digits) */
  primaryColor: PropTypes.string.isRequired,
  /** The background color for the secondary bar (checksum digit) */
  secondaryColor: PropTypes.string.isRequired
};

export default BarcodeChart;
