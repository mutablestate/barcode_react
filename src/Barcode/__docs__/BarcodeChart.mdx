---
name: BarcodeChart
route: /barcode/chart
menu: Barcode
---

import { Playground, Props } from 'docz';
import BarcodeChart from '../BarcodeChart';

# BarcodeChart

The generated barcode output represented by colored primary bars and a checksum bar.

## Props

<Props of={BarcodeChart} />

## Basic usage

<Playground>
  <BarcodeChart
    bars={[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]}
    checksumDigit={4}
    primaryColor={'red'}
    secondaryColor={'blue'}
    status={'Enter up to 19 digits'}
  />
</Playground>
