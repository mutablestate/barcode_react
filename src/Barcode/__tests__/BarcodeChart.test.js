import React from 'react';
import renderer from 'react-test-renderer';

import BarcodeChart from '../BarcodeChart';

const requiredProps = {
  bars: [1, 2, 3, 4],
  checksumDigit: 1,
  primaryColor: 'blue',
  secondaryColor: 'purple'
};

describe('BarcodeChart', () => {
  it('must render correctly with required props', () => {
    const component = renderer
      .create(<BarcodeChart {...requiredProps} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
