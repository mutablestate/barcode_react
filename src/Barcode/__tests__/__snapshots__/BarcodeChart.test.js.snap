// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarcodeChart must render correctly with required props 1`] = `
Array [
  <h4
    className="align-center"
  >
    Generated Barcode
  </h4>,
  <div
    className="flex-container"
  >
    <div
      data-test="BarcodeChart:Bar:div:blue"
      style={
        Object {
          "backgroundColor": "blue",
          "flex": "0 0 10px",
          "height": "20%",
        }
      }
    >
      1
    </div>
    <div
      data-test="BarcodeChart:Bar:div:blue"
      style={
        Object {
          "backgroundColor": "blue",
          "flex": "0 0 10px",
          "height": "30%",
        }
      }
    >
      2
    </div>
    <div
      data-test="BarcodeChart:Bar:div:blue"
      style={
        Object {
          "backgroundColor": "blue",
          "flex": "0 0 10px",
          "height": "40%",
        }
      }
    >
      3
    </div>
    <div
      data-test="BarcodeChart:Bar:div:blue"
      style={
        Object {
          "backgroundColor": "blue",
          "flex": "0 0 20px",
          "height": "50%",
        }
      }
    >
      4
    </div>
    <div
      data-test="BarcodeChart:Bar:div:purple"
      style={
        Object {
          "backgroundColor": "purple",
          "flex": "0 0 10px",
          "height": "20%",
        }
      }
    >
      1
    </div>
  </div>,
]
`;
